monai/data/image_writer.py

```diff
@@ -0,0 +1,395 @@
+        )
+        self.channel_dim = channel_dim
+
+    def set_metadata(self, meta_dict: Optional[Mapping] = None, resample: bool = True, **options):
+        """Resample ``self.dataobj`` if needed.  This method assumes ``self.data_obj`` is a 'channel-last' ndarray."""
+        original_affine, affine, spatial_shape = self.get_meta_info(meta_dict)
+        self.data_obj, self.affine = self.resample_if_needed(
+            data_array=self.data_obj,
+            affine=affine,
+            target_affine=original_affine if resample else None,
+            output_spatial_shape=spatial_shape,
+            mode=options.pop("mode", GridSampleMode.BILINEAR),
+            padding_mode=options.pop("padding_mode", GridSamplePadMode.BORDER),
+            align_corners=options.pop("align_corners", False),
+            dtype=options.pop("dtype", np.float64),
+        )
+
+    def write(self, filename_or_obj: PathLike, verbose: bool = False, **kwargs):
+        """Create an ITK object from ``self.data_obj`` and call ``itk.imwrite``"""
+        super().write(filename_or_obj, verbose=verbose)
+        self.data_obj = self.create_backend_obj(
+            self.data_obj, channel_dim=self.channel_dim, affine=self.affine, dtype=self.output_dtype, **kwargs  # type: ignore
+        )
+        itk.imwrite(
+            self.data_obj,
+            filename_or_obj,
+            compression=kwargs.pop("compression", False),
+            imageio=kwargs.pop("imageio", None),
+        )
+
+    @classmethod
+    def create_backend_obj(
+        cls,
+        data_array: NdarrayOrTensor,
+        channel_dim: Optional[int] = 0,
+        affine: Optional[NdarrayOrTensor] = None,
+        dtype: DtypeLike = np.float32,
+        **kwargs,
+    ):
+        """Create an ITK object from ``data_array``.  This method assumes a 'channel-last' ``data_array``."""
+        data_array = super().create_backend_obj(data_array)
+        _is_vec = channel_dim is not None
+        if _is_vec:
+            data_array = np.moveaxis(data_array, -1, 0)  # from channel last to channel first
+        data_array = data_array.T.astype(dtype, copy=True, order="C")
+        itk_obj = itk.GetImageFromArray(data_array, is_vector=_is_vec, ttype=kwargs.pop("ttype", None))
+
+        d = len(itk.size(itk_obj))
+        # convert affine to LPS
+        if affine is None:
+            affine = np.eye(d + 1, dtype=np.float64)
+        _affine: np.ndarray = convert_data_type(affine, np.ndarray)[0]  # type: ignore
+        _affine = cls.ras_to_lps(to_affine_nd(d, _affine))
+        spacing = np.sqrt(np.sum(np.square(_affine[:d, :d]), 0))
+        spacing[spacing == 0] = 1.0
+        _direction: np.ndarray = np.diag(1 / spacing)
+        _direction = _affine[:d, :d] @ _direction
+        itk_obj.SetSpacing(spacing.tolist())
+        itk_obj.SetOrigin(_affine[:d, -1].tolist())
+        itk_obj.SetDirection(itk.GetMatrixFromArray(_direction))
+        return itk_obj
+
+    @staticmethod
```

Comment:
May I know why using `classmethod` here? Maybe making it into a utility function directly?

Thanks.